sqlmesh/core/engine_adapter/bigquery.py

```diff
@@ -182,7 +182,32 @@ def query_factory() -> Query:
     def _begin_session(self, properties: SessionProperties) -> None:
         from google.cloud.bigquery import QueryJobConfig
 
-        job = self.client.query("SELECT 1;", job_config=QueryJobConfig(create_session=True))
+        query_label = []
+        if "query_label" in properties and isinstance(properties["query_label"], exp.Array):
+            for label_tuple in properties["query_label"].expressions:
+                if (
+                    isinstance(label_tuple, exp.Tuple)
+                    and len(label_tuple.expressions) == 2
+                    and all(isinstance(label, exp.Literal) for label in label_tuple.expressions)
+                ):
+                    query_label.append(
+                        (label_tuple.expressions[0].this, label_tuple.expressions[1].this)
+                    )
+                else:
+                    raise SQLMeshError(
+                        "Invalid query label format. Expected a tuple of two string literals."
```

Comment:
I feel like this is something that should be captured at model load time and not at execution time. This seems too late.

Reply:
The properties are already validated here: https://github.com/TobikoData/sqlmesh/blob/main/sqlmesh/core/model/common.py#L425-L432. Can we please extend that instead?